### PR TITLE
Use prebuilt binaries on Cairo installation

### LINKED ISSUE #388: Replace cairo installation from source with prebuilt binaries
Same as here: https://github.com/0xSpaceShard/cairo-cli-docker/issues/17

Perhaps we could do this in all test groups, not just configuration-tests (only installed there)

### PR DESCRIPTION
**Pull request by Nathan-SL**
## Usage related changes

-   NA
-   Closes #388

## Development related changes

-   Change Cairo source installation to using prebuilt binaries
-   Avail binaries for all test groups

## Checklist:

-   [x] Formatted the code
-   [x] No linter errors + tried to avoid introducing linter warnings
-   [x] Performed a self-review of the code
-   [x] Rebased to the last commit of the target branch (or merged it into my branch)
-   [ ] Documented the changes
-   [x] Updated the `test` directory (with a test case consisting of `network.json`, `hardhat.config.ts`, `check.ts`)
-   [x] Linked issues which this PR resolves
-   [x] All tests are passing (for external contributors who don't have access to the CI/CD pipeline)

### CODE REVIEW

File: test/general-tests/declare-v2-test/hardhat.config.ts [hunk 1]
```diff
@@ -2,7 +2,9 @@ import "@shardlabs/starknet-hardhat-plugin";
 
 module.exports = {
     starknet: {
-        network: process.env.NETWORK
+        venv: "active",
+        network: process.env.NETWORK,
+        cairo1BinDir: process.env.CAIRO_1_COMPILER_DIR
```
Comment (FabijanC) [inline test/general-tests/declare-v2-test/hardhat.config.ts:7]: Why change this?

File: scripts/setup-config-test-dependencies.sh [hunk 2]
```diff
@@ -2,34 +2,32 @@
...
+
+if [ -z "${CAIRO_1_COMPILER_DIR+x}" ]; then
+    # Setup cairo1 compiler
+    echo $COMPILER_BINARY_URL
+    mkdir -p cairo-compiler/target/release
+    curl --location -O --request GET "$COMPILER_BINARY_URL"
+    # Unzip asset and move to correct target
+    tar -zxvf $CAIRO_COMPILER_ASSET_NAME -C cairo-compiler --strip-components=1
+    mv cairo-compiler/bin/* cairo-compiler/target/release/
+    mv cairo-compiler/corelib cairo-compiler/target/corelib
```
Comment (FabijanC) [inline scripts/setup-config-test-dependencies.sh:26]: Why do we need preserve the corelib? Does it work without it?

File: scripts/setup-config-test-dependencies.sh [hunk 3]
```diff
@@ -2,34 +2,32 @@
 
 set -eu
 
-if [ "$TEST_SUBDIR" == "configuration-tests" ]; then
```
Comment (FabijanC) [inline scripts/setup-config-test-dependencies.sh:5]: The name of the script should be changed. I believe it only installs the compiler, so the name should reflect that.

**Review by FabijanC — CHANGES_REQUESTED** [review]

File: test/general-tests/declare-v2-test/hardhat.config.ts (hunk 1 above)
Comment (FabijanC) [reply test/general-tests/declare-v2-test/hardhat.config.ts:7]: I'm not sure. The compiler here is just a testing utility, we don't have to test a testing utility.

File: scripts/setup-config-test-dependencies.sh (hunk 2 above)
Comment (Nathan-SL) [reply scripts/setup-config-test-dependencies.sh:26]: If we miss `corelib` we'll get `Error: Failed to find development corelib.`

File: test/general-tests/declare-v2-test/hardhat.config.ts (hunk 1 above)
Comment (Nathan-SL) [reply test/general-tests/declare-v2-test/hardhat.config.ts]: https://github.com/0xSpaceShard/starknet-hardhat-plugin/pull/399/commits/c8f58ffedb82e9a1a794c733d17d31c40e80d2a7

File: scripts/setup-config-test-dependencies.sh (hunk 3 above)
Comment (Nathan-SL) [reply scripts/setup-config-test-dependencies.sh:5]: https://github.com/0xSpaceShard/starknet-hardhat-plugin/pull/399/commits/69563cf9a33d784fb285730482ae0a250ee3a7f4

**Review by FabijanC — APPROVED** [review]

**Pull request merged**